channeldb/invoices.go

```diff
@@ -1491,6 +1491,15 @@ func serializeInvoice(w io.Writer, i *Invoice) error {
 		return err
 	}
 
+	// Only if this is a _non_ AMP invoice do we serialize the HTLCs
+	// in-line with the rest of the invoice.
```

Comment:
Have you considered migrating the non AMP invoices to this format? A key such as `invoicenum | "regular"` could be used to store the htlcs. The advantage would be more uniform code. Disadvantage obviously is another migration to write.

Reply:
Hmm yeh main advantage is uniform code, migration is a big disadvantage imo, particularly as some nodes are now getting into the millions of invoices range.

Reply:
How much downtime would it be for those nodes?

Reply:
Depends on the the I/O they have access to really. If the node is running on a VPS with a low amount of provisioned iops, it could take in the 30 mins to 1 hour range for the initial update. Generally I think we should avoid _pure kv_ migrations that touch all invoices/payments, etc.